### DISCUSSION TITLE
Package size and read-only mode

### DISCUSSION THREAD

**Discussion by LiChoi** [react-page/react-page#941] [General] [answered]
Regarding read-only mode, this is what it says in the documentation:

"Readonly mode can be used for displaying the content to your users.
Simply provide a readOnly={true} prop to your editor.
We won’t load any UI related code unless readOnly changes to false. In this case, the ui will be lazy loaded if your bundler supports that (e.g. webpack)"

I'm not sure what is meant by UI related code. Is this the code that handles the page-editing functions (like all the form controls, dragging, dropping etc)?

https://www.npmjs.com/package/@react-page/editor package size is 2.59 MB. If I've npm installed this, won't the full package be sent to the browser regardless of readOnly? How does the readOnly option reduce the size of the code that is loaded?

If I only need to display the page that was built, and I don't need any editor-related code (since the page was already built at a different endpoint), is there a way to send just the "value" json of the page, and pass that into a slimmed down version of the editor component that only handles displaying?

**Comment by macrozone** [comment, marked as answer]
> If I only need to display the page that was built, and I don't need any editor-related code (since the page was already built at a different endpoint), is there a way to send just the "value" json of the page, and pass that into a slimmed down version of the editor component that only handles displaying?

that is what happens if `readOnly` is true.

> If I've npm installed this, won't the full package be sent to the browser regardless of readOnly? How does the readOnly option reduce the size of the code that is loaded?

Webpack does tree shake the package and only include what you actually use.
Additionaly, it handles "dynamic imports": `import("mypackage").then(exports => // use it)`. Packages imported like this are loaded on demand and not bundled

**Reply by LiChoi** [reply]
@macrozone  Thank you for clarifying!
